src/fragments/Graph.react.js

```diff
@@ -226,8 +226,27 @@ class PlotlyGraph extends Component {
         if (!layout) {
             return layout;
         }
-
-        return mergeDeepRight(layout, this.getLayoutOverride(responsive));
+        const override = this.getLayoutOverride(responsive);
+        const prev_layout = this.gd.current.layout; // === this.props.figure.layout
+        const prev_override_originals =
+            (this.state && this.state.override_originals) || {};
+        // Store the original data that we're about to override
+        const override_originals = {};
+        for (const key in override) {
+            override_originals[key] = layout[key];
```

Comment:
Mostly this seems right, but you're pulling out "originals" even if this is the same layout as before - which has already had the overrides applied to it.